components/external_server/src/model/issue_tracker.py

```diff
@@ -58,3 +94,16 @@ def _basic_auth_credentials(self) -> tuple[str, str] | None:
     def _auth_headers(self) -> dict[str, str]:
         """Return the authorization headers, if any."""
         return dict(Authorization=f"Bearer {self.private_token}") if self.private_token else {}
+
+
+def instantiate_issue_tracker(report: Report) -> IssueTracker:
```

Comment:
shouldn't this be a classmethod?
```suggestion
    @classmethod
    def from_report(cls, report: Report) -> IssueTracker:
        ...
        return cls(url, project_key, issue_type, username, password, private_token)
```

Reply:
or just use the __init__ method if this is the only way we initialize a issue_tracker. According the the dataclass docs you can write:
`@dataclass(init=False)`
to prevent it from creating an init method

Reply:
I was trying to prevent the issue tracker class from having knowledge about the internal structure of the report class. I think it makes more sense to ask a report for its issue tracker than passing a report to the issue tracker to have the issue tracker grab all the parameters from the report dict. On the other hand, I don't want to move the issue tracker class to the shared python code it it's only used in the external server. I'll subclass the report class in external server and add a get_issue_tracker() method there.